com.github.xournalpp.xournalpp.yaml

```diff
@@ -162,3 +157,95 @@ modules:
         x-checker-data:
           type: git
           tag-pattern: ^v([\d.]+)$
+
+  # Full version that is like 7 GB big
+  - name: texlive_extension
+    buildsystem: simple
+    build-commands:
+      - mkdir /app/extensions
+
+  # Smaller latex package
+  - name: tinytex
+    only-arches:
+      - x86_64
+    buildsystem: simple
+    build-commands:
+      - mkdir /app/tinytex
+      - tar xf TinyTeX-1-v2022.04.04.tar.xz -C /app/tinytex/ --strip-components 1
+
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ varwidth.zip
+      - /app/tinytex/bin/*/mktexlsr
+
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ scontents.zip
+      - cd /app/tinytex/texmf-dist/tex/latex/scontents/
+      - /app/tinytex/bin/*/latex -output-directory=/app/tinytex/texmf-dist/tex/latex/scontents scontents.ins
+      - /app/tinytex/bin/*/mktexlsr
+  
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ filemod.zip
+      - /app/tinytex/bin/*/mktexlsr
+  
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ gincltex.zip
+      - cd /app/tinytex/texmf-dist/tex/latex/gincltex/
+      - /app/tinytex/bin/*/latex -output-directory=/app/tinytex/texmf-dist/tex/latex/gincltex gincltex.ins
+      - /app/tinytex/bin/*/mktexlsr
+  
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ ydoc.zip
+      - cd /app/tinytex/texmf-dist/tex/latex/ydoc/
+      - /app/tinytex/bin/*/tex -output-directory=/app/tinytex/texmf-dist/tex/latex/ydoc ydoc.dtx
+      - /app/tinytex/bin/*/mktexlsr
+  
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ collectbox.zip
+      - cd /app/tinytex/texmf-dist/tex/latex/collectbox/
+      - /app/tinytex/bin/*/latex -output-directory=/app/tinytex/texmf-dist/tex/latex/collectbox collectbox.ins
+      - /app/tinytex/bin/*/mktexlsr
+  
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ adjustbox.zip
+      - cd /app/tinytex/texmf-dist/tex/latex/adjustbox/
+      - /app/tinytex/bin/*/latex -output-directory=/app/tinytex/texmf-dist/tex/latex/adjustbox adjustbox.ins
+      - /app/tinytex/bin/*/mktexlsr
+  
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ currfile.zip
+      - cd /app/tinytex/texmf-dist/tex/latex/currfile/
+      - /app/tinytex/bin/*/latex -output-directory=/app/tinytex/texmf-dist/tex/latex/currfile currfile.ins
+      - /app/tinytex/bin/*/mktexlsr
+  
+      - unzip -d /app/tinytex/texmf-dist/tex/latex/ standalone.zip
+      - cd /app/tinytex/texmf-dist/tex/latex/standalone/
+      - /app/tinytex/bin/*/latex -output-directory=/app/tinytex/texmf-dist/tex/latex/standalone standalone.ins
+      - /app/tinytex/bin/*/mktexlsr
+
+    sources:
+      - type: file
+        url: https://www.giuspen.net/software/TinyTeX-1-v2022.04.04.tar.xz
+        sha256: 3025930f9c9e989c0cac59826f8ee126d677aec728d20537b9e53e13ace29020
+      - type: file
+        url: https://mirrors.ctan.org/macros/latex/contrib/scontents.zip
```

Comment:
do more stable URLs exist? So that the build would have higher chances of working in two years from now.

Reply:
Some of the packages (standalone, collectbox, adjustbox, filemod, gincletex, filehook, ydoc) are available on https://github.com/MartinScharrer?tab=repositories.

Reply:
https://github.com/rstudio/tinytex-releases/releases/download/daily/TinyTeX-1.tar.gz

Reply:
yeah, well. the point is to be able to fetch today's resources in, say, two years, in order to have today's package build. So a fixed URL would be good where the returned content is expected to not change.

Reply:
I have replaced the URL's with stable versions and tried to keep it simple: Use the precompiled TinyTex (around 220 MB) containing [these packages](https://github.com/rstudio/tinytex/blob/main/tools/pkgs-yihui.txt) including `standalone` and `varwidth`, so that only the `scontents` package had to be added.
Alternatively `TinyTeX-1` could be used (around 66 MB), but then more packages would have to be added, in particular `standalone` and `varwidth`.